proposals/table-component.md

```diff
@@ -0,0 +1,221 @@
+      // logic to fetch data based on page number
+      // `currentPage` and `data` and `numPages` will be updated here.
+  }
+
+  componentDidMount = () => {
+      const {currentPage, pageSize} = this.state;
+      this.fetch(currentPage, pageSize);
+  }
+
+  onPageChange = (nextPage, prevPage) => {
+      const {pageSize} = this.state;
+      this.fetch(nextPage, pageSize);
+  }
+
+  render = () => {
+    return (
+        <Table
+          columns={columns}
+          dataSource={this.state.data}
+        />
+        <Pagination currentPage={this.state.currentPage} numPages={this.state.numPages} onPageChange={this.onPageChange}/>
+    );
+  }
+}
+```
+
+## Exports
+
+* `Table`
+* `StyledRoot`
+* `StyledHead`
+* `StyledHeadCell`
+* `StyledBody`
+* `StyledRow`
+* `StyledCell`
+
+## `Table` API
+
+* `columns: Array<{title: string, dataIndex: number, render?: (rowData: Array<any>) => React.Node, sortComparator?: (a: Array<any>, b: Array<any>) => number}>` - Required
+  * Describes how the column of data sorts and corresponding cells render.
+* `data: Array<Array<any>>` - Required
+  * Matrix data to be rendered within the table.
+* `estimatedRowSize?: number = 40px`
+  * Passed to `react-virtualized`. Used to estimate table height.
+* `overrides?: {Root, Head, HeadCell, Body, Row, Cell} = {}`
+  * `Root?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * Wraps the set of elements.
+  * `Head?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * Container for the table header related elements.
+  * `HeadCell?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * A header row. `Tr` for the `Thead` element.
+  * `Body?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * A header cell. Corresponds with the `th` html element.
+  * `Row?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * A data cell in the table. Corresponds with the `td` html element.
+  * `Cell?: ReactComponent | {props: {}, style: {}, component: ReactComponent}`
+    * A data row that contains `Td` components.
+* `useDynamicRowHeight?: boolean = false`
+  * Dynamically measure row heights, if size of rendered data is unknown.
+
+## `Column` API
+
+One of the Table columns prop for describing the table's columns, Column has the same API.
+
+* `title: string` - Required.
+  * Title of the column to display in the header.
+* `dataIndex?: number`
+  * Attribute at which to index the row data. Must provide a `sortComparator` or column may not be sorted.
+* `render: <T>(<T>, index, Array<T>) => $ReactNode`
+  * Customize the cells for this column.
+* `sortComparator?: (a: Array<any>, b: Array<any>`
+  * Defaults to [local sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) at provided `dataIndex`.
+
+## Dependencies
+
+* [react-virtualized](https://bvaughn.github.io/react-virtualized/#/components/List)
```

Comment:
should we look into https://github.com/bvaughn/react-window#frequently-asked-questions?

Reply:
i considered for quite a while if react-window was worth it. 

we would be pulling in a much smaller sized package.. but `react-virtualized` has some nice features (called out in the rfc) that `react-window` does not support.

Reply:
👍